Differentiate json@0.1.0

edsu7 opened the pull request.

edsu7 committed:
[wfpm v0.8.0] added starting template for differentiate-json@0.1.0

edsu7 committed:
added main files and test functions

edsu7 committed:
Update README.md

lindaxiang commented on differentiate-json/Dockerfile:
```diff
@@ -0,0 +1,27 @@
+FROM ubuntu:18.04
+
+LABEL org.opencontainers.image.source https://github.com/edsu7/argo-data-submission
```
https://github.com/icgc-argo/argo-data-submission

edsu7 replied:
Fixed
https://github.com/icgc-argo/argo-data-submission/blob/b4308cecdc24fa8d8760d8c3630856e22b972ac9/differentiate-json/Dockerfile#L3

lindaxiang commented on differentiate-json/main.py:
```diff
@@ -0,0 +1,126 @@
+
+    if len(warnings)>0:
+        with open('WARNINGS.log', 'w') as f:
+            for warning in warnings:
```
When will we get `WARNINGS.log`?

edsu7 replied:
I haven't coded for any `WARNING` scenarios yet.

lindaxiang commented on differentiate-json/main.py:
```diff
@@ -0,0 +1,126 @@
+        with open('ERRORS.log', 'w') as f:
+            for error in errors:
+                f.write(error+"\n")
+        raise ValueError(str(len(errors))+" errors detected. Please refer to ERRORS.log" )
```
How can we get the `ERRORS.log` from the tool?

edsu7 replied:
`ERRORS.log` will be saved within the working directory e.g. `work/a0/863e005fe3ea89d43a778e1aeff026/ERRORs.log` could otherwise output it as an emission?

lindaxiang replied:
We may not be able to get back the `ERRORS.log` as the step will fail as long as there are errors. :-)
I do not think this is an issue, as the workflow stops when there are errors is what we expect, right?

edsu7 replied:
Yeah the workflow stops when errors are detected. More-so I wanted the `ERRORS.log` to help inform why the workflow stopped.

lindaxiang commented on differentiate-json/main.py:
```diff
@@ -0,0 +1,126 @@
+    exceptions=[
+            "sample_barcode",
+            "insert_size",
+            "read_group_id_in_bam",
```
Why does it skip field `read_group_id_in_bam`?

edsu7 replied:
Ah sorry. That should be `submitter_read_group_id`. good catch

edsu7 replied:
Switched `read_group_id_in_bam` to `submitter_read_group_id`
https://github.com/icgc-argo/argo-data-submission/blob/b4308cecdc24fa8d8760d8c3630856e22b972ac9/differentiate-json/main.py#L52-L57

edsu7 committed:
correction in main.py and dockerfile

- `main.py`
-- from `read_group_id_in_bam` to 'switching `submitter_read_group_id`
-- adding `exceptions` to recursions
- `Dockerfile`
- fixing label

edsu7 commented on differentiate-json/main.py:
```diff
@@ -0,0 +1,128 @@
+
+                ###If key object ele is dictionary result in recursion
+                if type(entry[1])==dict:
+                    check_values(json_a[key][entry[0]],json_b[key][entry[0]],warnings,errors,exceptions,nested_key)
```
Fixed inner loop `check_values`
from : 
`check_values(json_a[key][entry[0]],json_b[key][entry[0]],warnings,errors,nested_key)`
to
`check_values(json_a[key][entry[0]],json_b[key][entry[0]],warnings,errors,exceptions,nested_key)`

lindaxiang replied:
good catch.
